### DISCUSSION TITLE
Why running a public instance is a bad idea?

### DISCUSSION THREAD

**Discussion by AdmiralAwesome** [bitmagnet-io/bitmagnet#282] [General]
Hey, I'm not planning to run any public instances of bitmagnet and I've seen one, already locked post about this topic (https://github.com/bitmagnet-io/bitmagnet/discussions/237).

My question is pretty simple - why do you advise not to run any public instances of bitmagnet? 

Is it because of security concerns? Or maybe because "_you are stealing from tracker sites_"? :D Or is there any other reason?

It's just an honest question and an open discussion, that is really all. Thanks!

**Comment by mgdigital** [comment]
Hey, I'm happy to expand on this in 2 parts.

Firstly, security: this is alpha software and (so far) has been built with only running a private instance in mind, either for purely personal use or trusted friends. If I have access to your instance I can perform destructive actions like deleting torrents and probably bring it down fairly simply with a denial of service or other type of attack. It's on the roadmap to improve security and provide authentication features and access levels that might make it more feasible to open up instances in some ways. Until then it would be irresponsible to encourage this here.

Secondly, the reason the other thread was shut down very quickly is to ensure compliance with GitHub's Terms of Service. This project has no control over content on the DHT or on users' bitmagnet instances, but it is responsible for moderating content posted in this repository. Breaking the TOS risks the future of this repo on GitHub, and the other thread was inviting people to do so. It was little different from "hey people, come and post links to the Pirate Bay here!".

Once the first part is improved, open instances might be more feasible but GitHub isn't going to be a good platform for advertising them (although there might be a locked down demo instance with some of my favourite Linux ISO's available at some point!) 

GitHub is a great platform for hosting open source projects though - let's ensure we can keep using it. Hope that makes sense :)

**Reply by ralyodio** [reply]
Do you know how many torrents are in the dht roughly and how long bitmagnet would take to index them all?  I'm at 2.4M in about two weeks so far

**Reply by becks0815** [reply]
> Do you know how many torrents are in the dht roughly and how long bitmagnet would take to index them all? I'm at 2.4M in about two weeks so far

I have seen screenshots with 40 Mio hashes. My instance is at 25 mio at the moment.

**Reply by ralyodio** [reply]
wow that's amazing. i'm about about 3M

**Reply by nojacko** [reply]
Not Bitmagnet but there's another public DHT crawler that claims to have 93,494,520 torrents.
